### PR TITLE
[PR] Continue build out of contact and faculty associations with factsheets

### PR DESCRIPTION
**Pull request by jeremyfelt**

### CODE REVIEW

**Comment by jeremyfelt** [comment]
@philcable #reviewmerge?

File: includes/class-wsuwp-graduate-degree-programs.php [hunk 1]
```diff
@@ -354,8 +595,8 @@ public function display_factsheet_primary_meta_box( $post ) {
 				}
 
 				?>
-					<script type="text/template" id="factsheet-deadline-template">
-						<span class="factsheet-<?php echo esc_attr( $meta['type'] ); ?>-field">
+				<script type="text/template" id="factsheet-deadline-template">
+					<span class="factsheet-<?php echo esc_attr( $meta['type'] ); ?>-field">
 							<select name="<?php echo esc_attr( $key ); ?>[<%= form_count %>][semester]">
```
Comment (philcable) [inline includes/class-wsuwp-graduate-degree-programs.php]: Certainly not critical, but it looks this and the following few lines should be outdented by one tab, too.

File: includes/class-wsuwp-graduate-degree-programs.php [hunk 2]
```diff
@@ -417,29 +658,26 @@ public function display_factsheet_primary_meta_box( $post ) {
 				}
 
 				?>
-					<script type="text/template" id="factsheet-requirement-template">
-						<span class="factsheet-<?php echo esc_attr( $meta['type'] ); ?>-field">
+				<script type="text/template" id="factsheet-requirement-template">
+					<span class="factsheet-<?php echo esc_attr( $meta['type'] ); ?>-field">
 							<input type="text" name="<?php echo esc_attr( $key ); ?>[<%= form_count %>][score]" value="" />
```
Comment (philcable) [inline includes/class-wsuwp-graduate-degree-programs.php]: These guys could be outdented too.

File: includes/class-wsuwp-graduate-degree-programs.php [hunk 3]
```diff
@@ -586,6 +824,49 @@ public function save_factsheet( $post_id, $post ) {
...
+				if ( in_array( $chair_selection['program_chair'], array( 'true', 'false' ), true ) ) {
+					$faculty_relationships[ $term_id ]['chair'] = $chair_selection['program_chair'];
+				} else {
+					$faculty_relationships[ $term_id ]['chair'] = 'false';
+				}
+
+				if ( in_array( $chair_selection['program_cochair'], array( 'true', 'false' ), true ) ) {
+					$faculty_relationships[ $term_id ]['cochair'] = $chair_selection['program_cochair'];
+				} else {
+					$faculty_relationships[ $term_id ]['chair'] = 'false';
```
Comment (philcable) [inline includes/class-wsuwp-graduate-degree-programs.php]: s/chair/cochair

File: includes/class-wsuwp-graduate-degree-programs.php [hunk 4]
```diff
@@ -270,52 +307,256 @@ public function display_factsheet_primary_meta_box( $post ) {
...
+	 *
+	 * @since 0.7.0
+	 *
+	 * @param WP_Post $post
+	 */
+	public function display_faculty_meta_box( $post ) {
+		$faculty_members = wp_get_object_terms( $post->ID, 'gs-faculty' );
+		$faculty_relationships = get_post_meta( $post->ID, 'gsdp_faculty_relationships', true );
+
+		foreach ( $faculty_members as $faculty_member ) {
```
Comment (philcable) [inline includes/class-wsuwp-graduate-degree-programs.php]: I'm probably just missing something, but even with a few Faculty terms, nothing shows up in this meta box for me.

File: includes/class-graduate-degree-contact-taxonomy.php [hunk 5]
```diff
@@ -174,95 +188,18 @@ public static function get_all_term_meta( $term_id ) {
...
-		<?php
+
+		foreach( $this->term_meta_keys as $key => $meta ) {
+			?>
+			<tr class="form-field">
+				<th scope="row">
+					<label for="<?php echo esc_attr( $key ); ?>"><?php echo esc_html( $meta['description'] ); ?></label>
+				</th>
+				<td>
+					<input type="text" name="<?php echo esc_attr( $key ); ?>" id="<?php echo esc_attr( $key ); ?>" value="<?php echo esc_attr( $term_meta[ $key ][0] ); ?>" />
```
Comment (philcable) [inline includes/class-graduate-degree-contact-taxonomy.php]: The `value` output here pops an `undefined index` notice when editing a newly-added contact.

File: includes/class-wsuwp-graduate-degree-programs.php (hunk 4 above)
Comment (philcable) [inline includes/class-wsuwp-graduate-degree-programs.php]: You already resolved this one 👍

File: includes/class-wsuwp-graduate-degree-programs.php [hunk 6]
```diff
@@ -270,52 +307,256 @@ public function display_factsheet_primary_meta_box( $post ) {
...
+			'textarea_rows' => 10,
+			'media_buttons' => false,
+			'teeny' => true,
+		);
+
+		if ( isset( $meta['pre_html'] ) ) {
+			echo $meta['pre_html']; // @codingStandardsIgnoreLine (HTML is static in code)
+		}
+		?>
+		<div class="factsheet-primary-input factsheet-<?php echo esc_attr( $meta['type'] ); ?>"">
```
Comment (jeremyfelt) [inline includes/class-wsuwp-graduate-degree-programs.php]: Doh. Good catch!

File: includes/class-wsuwp-graduate-degree-programs.php (hunk 3 above)
Comment (jeremyfelt) [inline includes/class-wsuwp-graduate-degree-programs.php]: That would have caused some interesting issues. :man_facepalming:

**Comment by jeremyfelt** [comment]
Thanks, @philcable! I've updated with a few changes. #reviewmerge again?

**Comment by philcable** [comment]
👍  Looks good!

**Pull request merged**